### PR TITLE
fix(es/compat): Insert loop Var into closest place

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**

- Closes: #8300

### CODE REVIEW

**Comment by Austaras** [comment]
This doesn't seem to be the best solution. Consider following example
```js
class Foo extends Bar {
    constructor() {
        for (const i of [1]) {
            setTimeout(() => {
                console.log(this)
            });
        }
        super();
    }
}
```

**Comment by magic-akari** [comment]
> This doesn't seem to be the best solution. Consider following example
> 
> ```js
> class Foo extends Bar {
>     constructor() {
>         for (const i of [1]) {
>             setTimeout(() => {
>                 console.log(this)
>             });
>         }
>         super();
>     }
> }
> ```

@Austaras Test cases updated. IMO, it's OK

**Comment by Austaras** [comment]
It's not. The problem arises in access this before call super.

File: crates/swc/tests/fixture/issues-8xxx/8300/output/3.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Foo extends Bar {
+    constructor(){
+        var _this = this, _loop = function(i) {
```
Comment (magic-akari) [inline crates/swc/tests/fixture/issues-8xxx/8300/output/3.js:3]: 🤔 This is a bug.

Comment (Austaras) [reply crates/swc/tests/fixture/issues-8xxx/8300/output/3.js:3]: > Uncaught ReferenceError: must call super constructor before using 'this' in derived class constructor

**Pull request closed without merging**